File: taste_frame/claculate_norm.py
import numpy as np

def claculate_norm(X,A,K,PARFOR_FLAG):
    #UNTITLED3 Summary of this function goes here
    #   Detailed explanation goes here
    normX = 0
    Size_input=(A.shape[0] * A.shape[1])
    num_non_z = np.count_nonzero(A)
    normA = np.sum(np.square(A))
    if PARFOR_FLAG:
        #parallel for loop
        for k in range(K):
            normX += np.sum(np.square(X[k]))
            Size_input += (X[k].shape[0] * X[k].shape[1])
            num_non_z += np.count_nonzero(X[k])
    else:
        for k in range(K):
            normX += np.sum(np.square(X[k]))
            Size_input += (X[k].shape[0] * X[k].shape[1])
            num_non_z += np.count_nonzero(X[k])

    return normX,normA,Size_input

File: legacy/TASTE_Framework/claculate_norm.m
function [ normX,normA,Size_input ] = claculate_norm(X,A,K,PARFOR_FLAG)
%UNTITLED3 Summary of this function goes here
%   Detailed explanation goes here
    normX=0;
    Size_input=(size(A,1)*size(A,2));
    num_non_z=nnz(A);
    normA=sum(sum( (A).^2));
    if (PARFOR_FLAG)
        parfor k = 1:K
         normX = normX + sum(sum( (X{k}).^2));
         Size_input=Size_input+(size(X{k},1)*size(X{k},2));
         num_non_z=num_non_z+nnz(X{k});
        end
    else
        for k = 1:K
         normX = normX + sum(sum( (X{k}).^2));
         Size_input=Size_input+(size(X{k},1)*size(X{k},2));
         num_non_z=num_non_z+nnz(X{k});
        end
    end


end

